src/prefect/utilities/filesystem.py

```diff
@@ -107,3 +108,13 @@ def to_display_path(
     relative_path = str(path.relative_to(relative_to))
     absolute_path = str(path)
     return relative_path if len(relative_path) < len(absolute_path) else absolute_path
+
+
+def platform_specific_relpath(path_str: str) -> Path:
```

Comment:
So this: "Converts a relative path generated on any platform to a relative path for the current platform"

I'm not sure this is captured by the name/docstring. Perhaps `relative_path_to_current_platform`? hmm.

Reply:
`Converts a relative path generated on any platform to a relative path for the current platform`. That's a very succinct way of putting it. I'll update the docstring accordingly. 

The other names I tried felt pretty kludgy. `platform_appropriate_relative_path` was the second choice I had.

Reply:
Maybe `path_to_current_platform`? Then just _recommend_ that it's a relative path in the docstring?

Reply:
My sticking point is that there's no amount of magic that will fix you passing `C:\Users\peytonrunyan\work\new prefect\prefect` as a path. It would need a `.resolve()` call.

Reply:
I can do `relative_path_to_current_platform` with the update docstring though

Reply:
That makes a lot of sense. I guess I'm back to my original suggestion :D 

My opposition to `platform_appropriate` is that it's not clear that it is converting the given path to work for the current platform.